File: java/L200/LeetCode0274.java
package L200;

import java.util.Arrays;

public class LeetCode0274 {
    public int hIndex(int[] citations) {
        if (citations == null || citations.length == 0) {
            return 0;
        }
        Arrays.sort(citations);
        for (int h = 0;h < citations.length;++h) {
            if (citations[h] >= citations.length - h) {
                return citations.length - h;
            }
        }
        return 0;
    }
}

File: cpp/L200/LeetCode0274.cc
#include <algorithm>
#include <vector>
using std::vector;

class Solution {
public:
    int hIndex(vector<int>& citations) {
        std::sort(citations.begin(), citations.end());
        for (int i = 0;i < citations.size();++i) {
            if (citations.size() - i <= citations[i]) {
                return citations.size() - i;
            }
        }
        return 0;
    }
};